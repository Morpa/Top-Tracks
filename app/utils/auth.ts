import dotenv from 'dotenv'

dotenv.config()

const state = {
  token: null,
  expires: 0
}

export const getToken = async () => {
  const now = Math.round(new Date().getTime() / 1000)
  const remaining = state.expires - now

  if (remaining <= 60) {
    state.token = null
    state.expires = 0
  }

  if (state.expires === 0) {
    const { access_token, expires_in } = await fetch(
      'https://accounts.spotify.com/api/token',
      {
        method: 'POST',
        headers: {
          Authorization: `Basic ${Buffer.from(
            process.env.SPOTIFY_CLIENT_ID +
              ':' +
              process.env.SPOTIFY_CLIENT_SECRET
          ).toString('base64')}`,
          'Content-Type': 'application/x-www-form-urlencoded'
        },
        body: 'grant_type=client_credentials'
      }
    ).then((response) => response.json())

    state.token = access_token
    state.expires = now + expires_in
  }

  return state.token
}
