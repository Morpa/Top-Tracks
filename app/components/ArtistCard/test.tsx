import { render, screen } from '~/utils/test-utils'

import { ArtistCard } from '.'

const props = {
  id: '1',
  title: 'Depeche Mode',
  image_url: 'https://i.scdn.co/image/ab6761610000e5eb8ada6bad4c9b276b7fd24874'
}

describe('<ArtistCard />', () => {
  it('should render correctly', () => {
    const { container } = render(<ArtistCard {...props} />)

    expect(
      screen.getByRole('img', {
        name: /depeche mode/i
      })
    ).toBeInTheDocument()

    expect(screen.getByText(props.title)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a link', () => {
    render(<ArtistCard {...props} />)

    expect(
      screen.getByRole('img', { name: props.title }).closest('a')
    ).toHaveAttribute('href', `artist/${props.id}`)
  })
})
