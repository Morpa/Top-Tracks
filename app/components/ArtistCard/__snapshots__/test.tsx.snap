// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistCard /> should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0.4rem;
}

.c1 {
  min-height: 14rem;
  position: relative;
  width: 100%;
  background: #f6f7f8;
}

.c2 {
  object-fit: cover;
  width: 100%;
  height: 19rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 100%;
  margin: 0.8rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #030517;
  font-size: 2.8rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 3.2rem;
  max-width: calc(100% - 2.5rem);
}

<a
  class="c0"
  href="/tracks/1"
>
  <div
    class="c1"
  >
    <img
      alt="Depeche Mode"
      class="c2"
      src="https://i.scdn.co/image/ab6761610000e5eb8ada6bad4c9b276b7fd24874"
    />
  </div>
  <div
    class="c3"
  >
    <span
      class="c4"
    >
      Depeche Mode
    </span>
  </div>
</a>
`;
